lib/terrafying/generator.rb

```diff
@@ -250,6 +250,15 @@ def template(relative_path, params = {})
 
     def output_with_children
       @children.inject(@output) { |out, c| out.deep_merge(c.output_with_children) }
+      out = @output
+      if @opts_provider
+        out['resource'].keys.each do |type|
+          out['resource'][type].keys.each do |id|
+            out['resource'][type][id]['provider'] = "aws.#{@opts_provider}"
```

Comment:
Why not just set the provider name to `@opts_provider` and then specify the provider on the outside as `provider: 'aws.my-provider'?

 That way this would work for other provider types instead of assuming it's always aws.